spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java

```diff
@@ -218,20 +214,7 @@ public void afterSingletonsInstantiated() {
 				+ " bean name must be " + MANAGEMENT_CONFIGURER_NAME);
 		if (ClassUtils.isPresent("io.micrometer.core.instrument.MeterRegistry",
 				IntegrationManagementConfigurer.class.getClassLoader())) {
-			try {
-				io.micrometer.core.instrument.MeterRegistry registry = this.applicationContext.getBean(
-						io.micrometer.core.instrument.MeterRegistry.class);
-				if (this.applicationContext instanceof GenericApplicationContext
-						&& !this.applicationContext.containsBean(MICROMETER_CAPTOR_NAME)) {
-					((GenericApplicationContext) this.applicationContext).registerBean(MICROMETER_CAPTOR_NAME,
-							MicrometerMetricsCaptor.class,
-							() -> new MicrometerMetricsCaptor(registry));
-					this.metricsCaptor = this.applicationContext.getBean(MicrometerMetricsCaptor.class);
-				}
-			}
-			catch (NoSuchBeanDefinitionException e) {
-				// no op
-			}
+			this.metricsCaptor = MicrometerMetricsCaptor.loadCaptor(this.applicationContext);
```

Comment:
Right, that is good fix. This way we isolate any Micrometer references from the places where its absence in the classpath may cause some problems.